hazelcast-client/src/test/java/com/hazelcast/client/executor/durable/ClientDurableExecutorServiceTest.java

```diff
@@ -188,6 +188,30 @@ public void onFailure(Throwable t) {
         assertFalse(onResponse.get());
     }
 
+    public void testFullRingBuffer_WithExecutionCallback() throws InterruptedException {
+        String key = randomString();
+        DurableExecutorService service = client.getDurableExecutorService(SINGLE_TASK + randomString());
+        service.submitToKeyOwner(new SleepingTask(100), key);
+        DurableExecutorServiceFuture<String> future = service.submitToKeyOwner(new BasicTestCallable(), key);
+        final CountDownLatch latch = new CountDownLatch(1);
+        future.andThen(new ExecutionCallback<String>() {
+            @Override
+            public void onResponse(String response) {
+            }
+
+            @Override
+            public void onFailure(Throwable t) {
+                if (t.getCause() instanceof RejectedExecutionException) {
+                    latch.countDown();
+                }
+            }
+        });
+        assertOpenEventually(latch, 10);
```

Comment:
`assertOpenEventually(latch, 10)` can fail under high load, in slow environment. I think better is to just leave default timeout...
`assertOpenEventually(latch)`